[Nullability Annotations to Java Classes] Use Updated and Null Proof `Media` Payload Classes (`safe`)

ParaskP7 opened the pull request:
Parent [FluxC#2798](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/2798)
Accompanying [FluxC#2878](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2878)

This PR accompanying the above FluxC PR, and uses the updated and null-proof `Media` payload classes.

FYI: This change is `safe`, meaning that there shouldn't be any (major) compile-time changes associated with this change.

-----

@AjeshRPai I added you as the main reviewer, not so randomly, due to the fact that you're also reviewing [FluxC#2878](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2878), and I wanted you from the Jetpack/WordPress mobile team again to be aware of and sign-off on that change for JP/WPAndroid. Feel free to merge this PR directly yourself if you deem so.

-----

FluxC Update List:

1. [Update fluxc version to pr hash (2878)](https://github.com/wordpress-mobile/WordPress-Android/commit/1e85f08f59865888c1c5f04259e805c7e303d32a)
2. [Update fluxc version to trunk hash (2878)](https://github.com/wordpress-mobile/WordPress-Android/pull/19428/commits/2e605ea9b0b40a1c7712a600931f99810b40cfd5)

Test List:

1. [Update post media handler test to updated media payload](https://github.com/wordpress-mobile/WordPress-Android/commit/966023e8c54651915d24f112da7f42f05d2de213)

-----

## To Test (`REST`):

- Smoke test any media related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected. For a couple of examples, you can expand and follow the inner and more explicit test steps within:

## To Test (`XMLRPC`):

- Create a new self-hosted WordPress site for XMLRPC testing purposes ([jurassic.ninja](https://jurassic.ninja/)).
- Smoke test any media related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected. For a couple of examples, you can expand and follow the inner and more explicit test steps within:

<details>
    <summary>Media Screens [MediaBrowserActivity.java + MediaGridFragment.kt + MediaSettingsActivity.java + MediaPreviewActivity.java + MediaPreviewFragment.java]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.

- Go to `Media` screen, verify it is shown and functioning as expected.
- For example try changing tabs, searching and adding a new media.
- Tap on any media and verify that the `Media Settings` screen is shown and functioning as expected.
- For example try updating the media title or adding a media description.
- Tap on the media preview and verify that the `Media Preview` screen is shown and functioning as expected.
- For example try swipe left-right to navigate between media previews.

</details>

-----

# Merge instructions:

1. [x] Wait for the accompanying [FluxC#2878](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2878) PR to be merged.
3. [x] Update `wordPressFluxCVersion` to point to the `trunk` hash that includes the above solution.
4. [x] Remove the `[Status] Not Ready for Merge` label.
5. [x] Merge this PR.

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

ParaskP7 committed:
Build: Update fluxc version to pr hash (#2878)

This 'FluxC' PR hash updates the library to that branch version
where the 'Media' payload classes are updated to their new null proof
version.

FluxC PR: https://github.com/wordpress-mobile/
WordPress-FluxC-Android/pull/2878

This step is required in order to check that these 'FluxC'
related changes work as expected for WPAndroid.

ParaskP7 committed:
Test: Update post media handler test to updated media payload

AjeshRPai reviewed (approved):
Smoke tested the media screen in the App, everything looks good to me

ParaskP7 commented:
Great, thanks for reviewing and testing this @AjeshRPai , I am glad that everything is working as expected! 🙇 ❤️ 🚀

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into analysis/use-updated-and-null-proff-media-payloadl-classes

ParaskP7 committed:
Build: Update fluxc version to trunk hash (#2878)

This 'FluxC' hash updates the library to that 'trunk' version
where the 'Media' payload classes are updated to their new null proof
version.

FluxC PR: https://github.com/wordpress-mobile/
WordPress-FluxC-Android/pull/2878
